src/Microsoft.DotNet.Interactive.HttpRequestParser/Parser/HttpDiagnostics.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.DotNet.Interactive.HttpRequest;
+
+internal static class HttpDiagnostics
+{
+    internal static HttpDiagnosticInfo GetDiagnosticInfo(
+        this WellKnownHttpDiagnostics diagnosticCode,
+        params object[] messageArguments)
+    {
+        var id = $"HTTP{(int)diagnosticCode:0000}";
+        string messageFormat;
+        var severity = DiagnosticSeverity.Error;
+
+        switch (diagnosticCode)
+        {
+            case WellKnownHttpDiagnostics.UnrecognizedVerb:
+                messageFormat = "Unrecognized HTTP verb '{0}'.";
```

Comment:
I would strongly recommend creating a class that exposes each message as a method so that understanding the number of tokens that need to be passed in the array isn't an issue.